### PR TITLE
Sg modify scope by port

### PR DESCRIPTION
**Pull request by harelsa**
A rework for  an existing bot that made for Amdocs

### CODE REVIEW

File: bots/Bots.md [hunk 1]
```diff
@@ -405,6 +409,58 @@ arn:aws:s3:::my-bucket/my-logs/
...
+
+Example: AUTO: s3_block_public_access <true|false|-> <true|false|->
+When - set for ignore the specific attribute if its optional so usage will be
+
+Limitations: none
+
+Notes:
+    -  before running this bot, ensure that your applications will work correctly without public access
+    
+Another Examples:
```
Comment (RavidAtias) [inline bots/Bots.md:426]: Please ask Guy to go over those examples

File: bots/Bots.md [hunk 2]
```diff
@@ -31,6 +31,10 @@
   - [sg\_single\_rule\_delete](#sg_single_rule_delete)
   - [tag\_ec2\_resource](#tag_ec2_resource)
   - [vpc\_turn\_on\_flow\_logs](#vpc_turn_on_flow_logs)
+  - [s3\_block\_public\_access](#s3_block_public_access)
```
Comment (RavidAtias) [inline bots/Bots.md:34]: You don't work correctly with git, you have here 2 bots but you wanted to add only one to this pr.
Feel free to contact if you miss something, it is very simple

File: bots/sg_single_rule_delete.py [hunk 3]
```diff
@@ -1,362 +0,0 @@
-'''
```
Comment (RavidAtias) [inline bots/sg_single_rule_delete.py:1]: Why you deleted it?

File: bots/sg_modify_scope_by_port.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+                IpProtocol=rule[PROTOCOL].lower()
+            )
+            text_output = text_output + ' rule : ' + rule[SCOPE] + ',' + str(rule[PORT_FROM]) + ',' + str(
+                rule[PORT_TO]) + ',' + \
+                          str(sg_id) + ',' + rule[PROTOCOL].lower() + ' deleted successfully ;'
+
+        except Exception as e:
+            text_output = text_output + f'Error while trying to delete security group. Error: {e}'
+
+        try:
```
Comment (RavidAtias) [inline bots/sg_modify_scope_by_port.py:86]: try should wrap the api call

File: bots/sg_modify_scope_by_port.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+                IpPermissions=[  # only IpPermissions supported with this func !
+                    {
+                        'FromPort': rule[PORT_FROM],
+                        'IpProtocol': rule[PROTOCOL].lower(),
+                        'IpRanges': [
+                            {
+                                'CidrIp': scope
+                            },
+                        ],
+                        # 'Ipv6Ranges': [ # future work
```
Comment (RavidAtias) [inline bots/sg_modify_scope_by_port.py:147]: Delete comments

File: bots/sg_modify_scope_by_port.py [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+            text_output = text_output + ' rule : ' + rule[SCOPE] + ',' + str(rule[PORT_FROM]) + ',' + str(
+                rule[PORT_TO]) + ',' + \
+                          str(sg_id) + ',' + rule[PROTOCOL].lower() + ' deleted successfully ;'
+
+        except Exception as e:
+            text_output = text_output + f'Error while trying to delete security group. Error: {e}'
+
+        try:
+            # check if one is existing first ! avoid duplicates and exception
+            if rule_exists(sg, rule, direction, scope):
```
Comment (RavidAtias) [inline bots/sg_modify_scope_by_port.py:88]: Avoid using code inside if/else and create function with explicit name so the code will be readable

File: bots/sg_single_rule_delete.py (hunk 3 above)
Comment (harelsa) [reply bots/sg_single_rule_delete.py:1]: I will create new pull request only for this change

File: bots/Bots.md (hunk 2 above)
Comment (harelsa) [reply bots/Bots.md:34]: done

File: bots/sg_modify_scope_by_port.py (hunk 4 above)
Comment (harelsa) [reply bots/sg_modify_scope_by_port.py:86]: I kept only the api call in the try scope and moved the rest (if condition - rule_exists) out of it

File: bots/sg_modify_scope_by_port.py (hunk 5 above)
Comment (harelsa) [reply bots/sg_modify_scope_by_port.py:147]: done

File: bots/sg_modify_scope_by_port.py (hunk 6 above)
Comment (harelsa) [reply bots/sg_modify_scope_by_port.py:88]: code moved out of the if, and the name changed to is_rule_exists_in_sg

File: bots/Bots.md (hunk 1 above)
Comment (harelsa) [reply bots/Bots.md:426]: Guy instructed me through the bot changes and sent the bot to the client himself after approving it, and the usage ofcourse.

File: bots/sg_single_rule_delete.py (hunk 3 above)
Comment (RavidAtias) [reply bots/sg_single_rule_delete.py:1]: ?

Comment (harelsa) [reply bots/sg_single_rule_delete.py:1]: I hope now it's fixed after my commit

File: bots/Bots.md (hunk 1 above)
Comment (harelsa) [reply bots/Bots.md]: fixed

File: bots/sg_single_rule_delete.py (hunk 3 above)
Comment (harelsa) [reply bots/sg_single_rule_delete.py]: I Reverted the delete

File: bots/cloudtrail_enable_log_file_validation.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
+"""
+## cloudtrail_enable_log_file_validation
+What it does: Enable log file validation in cloudTrail
+Usage: AUTO: cloudtrail_enable_log_file_validation
+Limitations: None
+"""
```
Comment (RavidAtias) [inline bots/cloudtrail_enable_log_file_validation.py:6]: 6 files change in this PR.. something went wrong, lets do another quick session to fix everything if you are having problems

**Pull request closed without merging**